### DISCUSSION TITLE
How to add a MDX component to the title

### DISCUSSION THREAD

**Discussion by kayx23** [facebook/docusaurus#9158] [Q&A] [answered]
Hello, I made a reusable component - a button with tooltip - and would like to sometimes import and put it next to the title of the document for labelling. But the title is in the frontmatter of the markdown file. How would I go about this?

**Comment by slorber** [comment]
I don't understand sorry 😅 please show an Excalidraw mockup or something to show what you try to do.

You can swizzle any theme component and customize things this way

**Comment by kayx23** [comment]
In a markdown file, I can use a custom-built component like this, which adds it to the body of the document:

```markdown
---
title: Title
slug: /url/path
---

import Label from '@site/src/Components/Label';

<Label />
```

However, I want the component to be next to the title, such as:

```markdown
---
title: Title  <Label />
slug: /url/path
---
```

Rendered here:

![image](https://github.com/facebook/docusaurus/assets/39619599/5d7f8523-7a1e-42b3-b7b5-6792e5186a2c)

But obviously I cannot simply just put the ` <Label />` tag next to the title. 

I hope this is clear!

**Reply by slorber** [reply, marked as answer]
You can try something like this:

```mdx
---
title: Tutorial Intro
sidebar_label: Tutorial Intro
sidebar_position: 1
---

# Tutorial Intro <span style={{color: "red"}}>Hey</span>
```

<img width="436" alt="CleanShot 2023-07-21 at 09 51 01@2x" src="https://github.com/facebook/docusaurus/assets/749374/e74ffe7d-4675-44a2-abad-e132047e6963">


---

Make sure to use `sidebar_label` because otherwise, you will get the wrong text in the page title metadata, sidebar and other places like the breadcrumb and pagination buttons, because those elements are using plain strings and not MDX: your title JSX will be rendered as plain text inside those elements.

![CleanShot 2023-07-21 at 09 51 20@2x](https://github.com/facebook/docusaurus/assets/749374/a6e15e06-e9a0-4d87-9b9c-2fcd6db06221)

**Reply by kayx23** [reply]
thanks, super helpful!

**Comment by kayx23** [comment]
@slorber there's another issue. The tag has shown up in the `title` on the tab, such as

![image](https://github.com/facebook/docusaurus/assets/39619599/8fb60639-62a1-4d39-ac85-8fcc60badd36)

**Reply by kayx23** [reply]
Added the `title` in the frontmatter back and it resolved it. I previously removed it to go around a CI check that errorred out with two `<h1>`. Guess I have to live with two `<h1>` :?

**Reply by slorber** [reply]
You don't have 2 h1, you have a single one in the DOM. The DOM h1 use in priority the one you declare in markdown, while the frontmatter title is used in priority for docs metadata. Those 2 are decorelated notably for supporting cases like this where both need to be different.
